runtime/lua/vim/keymap.lua

```diff
@@ -79,6 +79,9 @@ function keymap.set(mode, lhs, rhs, opts)
   end
 end
 
+---@class KeymapDelOpts
```

Comment:
Make sure to namespace any classes you add.

Reply:
I had actually done that [initially](https://github.com/neovim/neovim/compare/b7453bf08af2634a43ce972c353ea61e97f80ef5..49cb777a91978a78656d5afaa7315bc84b163653), but the `doxygen` script is messed up and adds extra spaces that make CI fail :/

Reply:
Then we need to fix that or hold off on adding types that are exposed.

Reply:
Btw one trick I've used to exclude types from the docs is to use `--- @cast opts Myopts` at the top of the function body.